.github/CONTRIBUTING.md

```diff
@@ -31,141 +31,455 @@ strongly recommended you get approval/traction for it from our forums before sta
+Note, the ```to``` keyword is inclusive, so it automatically defaults to replacing ```<=```; if you want ```<``` then you should write it as ```1 to 
+some_value-1```).
+
+HOWEVER, if either ```some_value``` or ```i``` changes within the body of the for (underneath the ```for(...)``` header) or if you are looping over a list AND 
+changing the length of the list then you can NOT use this type of for-loop!
+
+### for(var/A in list) VS for(var/i in 1 to list.len)
+The former is faster than the latter, as shown by the following profile results:
+https://file.house/zy7H.png
+Code used for the test in a readable format:
+https://pastebin.com/w50uERkG
+
+#### Istypeless for loops
+A name for a differing syntax for writing for-each style loops in DM. It's NOT DM's standard syntax, hence why this is considered a quirk. Take a look at this:
+```DM
+var/list/bag_of_items = list(sword, apple, coinpouch, sword, sword)
+var/obj/item/sword/best_sword
+for(var/obj/item/sword/S in bag_of_items)
+  if(!best_sword || S.damage > best_sword.damage)
+    best_sword = S
+```
+The above is a simple proc for checking all swords in a container and returning the one with the highest damage, and it uses DM's standard syntax for a 
+for-loop by specifying a type in the variable of the for's header that DM interprets as a type to filter by. It performs this filter using ```istype()``` (or 
+some internal-magic similar to ```istype()``` - this is BYOND, after all). This is fine in its current state for ```bag_of_items```, but if ```bag_of_items``` 
+contained ONLY swords, or only SUBTYPES of swords, then the above is inefficient. For example:
+```DM
+var/list/bag_of_swords = list(sword, sword, sword, sword)
+var/obj/item/sword/best_sword
+for(var/obj/item/sword/S in bag_of_swords)
+  if(!best_sword || S.damage > best_sword.damage)
+    best_sword = S
+```
+specifies a type for DM to filter by. 
+
+With the previous example that's perfectly fine, we only want swords, but here the bag only contains swords? Is DM still going to try to filter because we gave 
+it a type to filter by? YES, and here comes the inefficiency. Wherever a list (or other container, such as an atom (in which case you're technically accessing 
+their special contents list, but that's irrelevant)) contains datums of the same datatype or subtypes of the datatype you require for your loop's body,
+you can circumvent DM's filtering and automatic ```istype()``` checks by writing the loop as such:
+```DM
+var/list/bag_of_swords = list(sword, sword, sword, sword)
+var/obj/item/sword/best_sword
+for(var/s in bag_of_swords)
+  var/obj/item/sword/S = s
+  if(!best_sword || S.damage > best_sword.damage)
+    best_sword = S
+```
+Of course, if the list contains data of a mixed type then the above optimisation is DANGEROUS, as it will blindly typecast all data in the list as the 
+specified type, even if it isn't really that type, causing runtime errors.
```

Comment:
You may want a bit harsher wording here. Just saying it's a runtime error gets the point across to more technical individuals, but for those who are new, it may need to be stressed a bit more how bad this is. *shrug*

Reply:
"DON'T DO THIS."